Sources/MongoSwift/MongoCursor.swift

```diff
@@ -6,16 +6,20 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
     private var _client: MongoClient?
     private var swiftError: Error?
 
+    /// Decoder from the `MongoCollection` or `MongoDatabase` that created this cursor.
+    private let decoder: BSONDecoder
+
     /**
      * Initializes a new `MongoCursor` instance, not meant to be instantiated directly.
      *
      * - Throws:
      *   - `UserError.invalidArgumentError` if the options passed to the command that generated this cursor formed an
      *     invalid combination.
      */
-    internal init(fromCursor: OpaquePointer, withClient: MongoClient) throws {
+    internal init(fromCursor: OpaquePointer, withClient: MongoClient, withDecoder: BSONDecoder) throws {
```

Comment:
this signature should probably be
`fromCursor cursor: OpaquePointer, withClient client: MongoClient, withDecoder decoder: BSONDecoder`

Reply:
fixed